cli/generate/src/build_tables/build_parse_table.rs

```diff
@@ -55,6 +55,7 @@ struct ParseTableBuilder<'a> {
     syntax_grammar: &'a SyntaxGrammar,
     lexical_grammar: &'a LexicalGrammar,
     variable_info: &'a [VariableInfo],
+    subtype_map: &'a Vec<(Symbol, Vec<ChildType>)>,
```

Comment:
After thinking about this, why do we map the symbol to `ChildType`? Like I mentioned earlier, conceptually, it'd make more sense to map the supertype symbol ids to the subtype symbol ids, thus meaning we don't actually need the symbol names. That indirection should happen outside of this relationship, via `ts_language_symbol_name`, so in here, I would think we would store a mapping of the symbol to an array of symbol ids that can be a potential subtype to the given supertype. With that in mind, please refactor this, it should clean up this section of the code quite a bit :) as I was really confused by the change in signature of `generate_node_types_json` to return the `ChildType` info, but really, we don't need that data.